From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Fri, 3 Apr 2020 18:08:28 +0000
Subject: [PATCH] Add IPPROTO_ETHERNET and IPPROTO_MPTCP from Linux 5.6 to
 netinet/in.h.

This patch adds the IPPROTO_ETHERNET and IPPROTO_MPTCP constants from
Linux 5.6 to glibc's netinet/in.h.

Tested for x86_64.

(cherry picked from commit f9ac84f92f151e07586c55e14ed628d493a5929d)

diff --git a/inet/netinet/in.h b/inet/netinet/in.h
index f94b4b4e1cc425d4f2ec0c3b59658e04d0490839..d44bf9c7b9bc2e23c4f11788d2941ea15648b5b8 100644
--- a/inet/netinet/in.h
+++ b/inet/netinet/in.h
@@ -87,8 +87,12 @@ enum
 #define IPPROTO_UDPLITE		IPPROTO_UDPLITE
     IPPROTO_MPLS = 137,    /* MPLS in IP.  */
 #define IPPROTO_MPLS		IPPROTO_MPLS
+    IPPROTO_ETHERNET = 143, /* Ethernet-within-IPv6 Encapsulation.  */
+#define IPPROTO_ETHERNET	IPPROTO_ETHERNET
     IPPROTO_RAW = 255,	   /* Raw IP packets.  */
 #define IPPROTO_RAW		IPPROTO_RAW
+    IPPROTO_MPTCP = 262,   /* Multipath TCP connection.  */
+#define IPPROTO_MPTCP		IPPROTO_MPTCP
     IPPROTO_MAX
   };
 
